packages/@react-aria/numberfield/src/useNumberField.ts

```diff
@@ -99,6 +100,16 @@ export function useNumberField(props: AriaNumberFieldProps, state: NumberFieldSt
     }
   });
 
+  let numberFormatter = useNumberFormatter(formatOptions);
+  let intlOptions = useMemo(() => numberFormatter.resolvedOptions(), [numberFormatter]);
+
+  // Replace negative textValue formatted using currencySign: 'accounting'
+  // with a textValue that can be announced using a minus sign.
+  let textValue =
+    intlOptions.style = 'currency' && intlOptions.currencySign === 'accounting' ?
+    inputValue.replace(/^\((.+)\)$/, '\u2212$1') :
```

Comment:
You could create a separate formatter for the `textValue` that doesn't use accounting and then you wouldn't need this regex. That might be more resilient if for some reason some locales use different formatting for accounting notation.

```js
let textValueFormatter = useNumberFormatter({...formatOptions, currencySign: undefined});
let textValue = useMemo(() => isNaN(state.numberValue) ? '' : textValueFormatter.format(state.numberValue), [textValueFormatter, state.numberValue]);
```